spec/features/dashboard/work_version_form_spec.rb

```diff
@@ -68,7 +69,22 @@
         expect(new_work_version.source).to eq [metadata[:source]]
 
         expect(page).to have_current_path(dashboard_form_contributors_path('work_version', new_work_version))
-        expect(SolrIndexingJob).not_to have_received(:perform_now)
+        expect(SolrIndexingJob).to have_received(:perform_now).at_least(:once)
+      end
+    end
+
+    context 'when saving-and-continuing, then hitting cancel' do
+      it 'indexes the work the initial time' do
+        visit dashboard_form_work_versions_path
+
+        FeatureHelpers::DashboardForm.fill_in_work_details(metadata)
+        FeatureHelpers::DashboardForm.save_and_continue
+
+        FeatureHelpers::DashboardForm.cancel
+
+        visit dashboard_root_path
+
+        expect(page).to have_content metadata[:title]
```

Comment:
So here there's no expectation check because we want it to actually index an check it in the dashboard?

Reply:
Yeah I figured here we should actually to a real blackbox end-to-end test to truly replicate the bug, rather than say "meh, that method was called so good enough ¯\_(ツ)_/¯"